### DISCUSSION TITLE
Topdown style RTS/4x camera?

### DISCUSSION THREAD

**Discussion by Abrinsky** [ramokz/phantom-camera#216] [Question / Help Wanted] [answered]
I'm posing this as more of a question than a feature request because I am not super familiar yet working with godot's camera, and I have been looking at phantom for a bit. 

two parts because one may answer the other:
I am looking for solutions to get a camera similar to an RTS or 4X strategy game, such that mouse against frame would move the camera in that direction, but also using input keys would move the camera as well.

Is this something phantom camera is already good at or would it be just as much work as just adding those features via script to the vanilla camera? (from what I've seen, none of the features really capture what I'm looking for except maybe frame 2d if the mouse was the "object") 

If not, is this an intended route for a feature you might add in the future? (This is more out of curiosity as I haven't found any addon that really delves much into this area)

I'm passionate about this because I have a barely working prototype using vanilla, and it just feels very sloppy (skill issue) but I am trying to get a well polished implementation.

**Comment by ramokz** [comment, marked as answer]
So input control for the `PCams` is not an intended feature for the addon. Main reason is that controlling a camera is often fairly specific depending on the project, and so introducing a solution to that would likely be a never ending task.

That said, think what you're looking for is fairly straightforward to implement still. What might be the easiest option would be to have a pseudo `follow target`. Basically, an invisible `Node2D`/`Node3D`, where you essentially modify the position of that target when your cursor gets close to the edge of the screen or when you press a particular key input.

There is a [Reddit post](https://www.reddit.com/r/godot/comments/pmd21r/edge_scrolling_in_2d_game/) that has an example for moving a target, in their case a `Camera2D`, whenever the mouse nears the edge of the screen. Which would be a way to then move the pseudo target, which in turn would move the `PCam` and `Camera2D/3D`.

**Reply by Abrinsky** [reply]
I'll definitely check this out, thanks for the thorough reply! Much appreciated and I think it answers a lot of what I was looking for.
